shared/domains/chelonia/time-sync.js

```diff
@@ -0,0 +1,95 @@
+import sbp from '@sbp/sbp'
+
+let wallBase = Date.now()
+let monotonicBase = performance.now()
+// `undefined` means the sync process has been stopped, `null` that the current
+// request has finished
+let resyncTimeout
+let watchdog
+
+const syncServerTime = async function () {
+  // Get our current monotonic time
+  const newMonotonicBase = performance.now()
+  // Now, ask the server for the time
+  const time = await fetch(`${this.config.connectionURL}/time`, { signal: this.abortController.signal })
+  const requestTimeElapsed = performance.now()
+  if (requestTimeElapsed - newMonotonicBase > 1000) {
+    throw new Error('Error fetching server time: request took too long')
+  }
+  // If the request didn't succeed, report it
+  if (!time.ok) throw new Error('Error fetching server time')
+  const serverTime = (new Date(await time.text())).valueOf()
```

Comment:
If it takes 200 ms for our request to get to the server, and 200ms to get the response, by the time we get the response from the server, the time we receive from the server will be 200ms old. So we should add 200ms to the time received from the server.

We can estimate this time by doing `(requestTimeElapsed - newMonotonicBase) / 2`